Userland/Libraries/LibCompress/Brotli.cpp

```diff
@@ -0,0 +1,905 @@
+
+            for (size_t rep = 0; rep < (repeat_count - last_repeat); rep++) {
+                result_symbols.append(i);
+                result_lengths.append(previous_non_zero_code_length);
+                result_lengths_count[previous_non_zero_code_length]++;
+
+                i++;
+
+                if (previous_non_zero_code_length != 0) {
+                    sum += (32768 >> previous_non_zero_code_length);
+                    if (sum == 32768)
+                        break;
+                    VERIFY(sum < 32768);
+                }
+            }
+            if (sum == 32768)
+                break;
+            VERIFY(sum < 32768);
+
+            last_repeat = repeat_count;
+        } else if (symbol == 17) {
+            size_t repeat_count = 0;
+            if (last_symbol == 17 && last_repeat != 0) {
+                repeat_count = (8 * (last_repeat - 2));
+            } else {
+                last_repeat = 0;
+            }
+            repeat_count += 3 + TRY(m_input_stream.read_bits(3));
+
+            i += (repeat_count - last_repeat);
+            last_repeat = repeat_count;
+        }
+
+        last_symbol = symbol;
+    }
+    result_lengths_count[0] = 0;
+
+    size_t code_value = 0;
+    for (size_t bits = 1; bits < 16; bits++) {
+        code_value = (code_value + result_lengths_count[bits - 1]) << 1;
+        size_t current_code_value = code_value;
+
+        for (size_t n = 0; n < result_symbols.size(); n++) {
+            size_t len = result_lengths[n];
+            if (len == bits) {
+                code.m_symbol_codes.append((1 << bits) | current_code_value);
+                code.m_symbol_values.append(result_symbols[n]);
+                current_code_value++;
+            }
+        }
+    }
+
+    return {};
+}
+
+static void inverse_move_to_front_transform(Span<u8> v)
+{
+    // RFC 7932 section 7.3
+    u8 mtf[256];
+    for (size_t i = 0; i < 256; ++i) {
+        mtf[i] = (u8)i;
+    }
+    for (size_t i = 0; i < v.size(); ++i) {
+        u8 index = v[i];
+        u8 value = mtf[index];
+        v[i] = value;
+        for (; index; --index) {
+            mtf[index] = mtf[index - 1];
+        }
+        mtf[0] = value;
+    }
+}
+
+ErrorOr<void> BrotliDecompressionStream::read_context_map(size_t number_of_codes, Vector<u8>& context_map, size_t context_map_size)
+{
+    bool use_run_length_encoding = TRY(m_input_stream.read_bit());
+    size_t run_length_encoding_max = 0;
+    if (use_run_length_encoding) {
+        run_length_encoding_max = 1 + TRY(m_input_stream.read_bits(4));
+    }
+    // dbgln("run_length_encoding_max: {}", run_length_encoding_max);
+
+    BrotliDecompressionStream::CanonicalCode code;
+    TRY(read_prefix_code(code, number_of_codes + run_length_encoding_max));
+
+    size_t i = 0;
+    while (i < context_map_size) {
+        // dbgln("i: {}", i);
+        size_t symbol = code.read_symbol(m_input_stream).release_value();
```

Comment:
```suggestion
        size_t symbol = TRY(code.read_symbol(m_input_stream));
```
Test file:
[crash-0834143266645039e9ceb21619ffa94920ceb4cb.txt](https://github.com/SerenityOS/serenity/files/8421893/crash-0834143266645039e9ceb21619ffa94920ceb4cb.txt)